osdep/semaphore_osx.c

```diff
@@ -77,18 +78,20 @@ int mp_sem_timedwait(mp_sem_t *sem, const struct timespec *abs_timeout)
         if (abs_timeout) {
             timeout_ms = 0;
```

Comment:
Do timeout_ns and use mp_poll bellow instead?

Reply:
macOS doesn't support ppoll or any equivalent to my knowledge.

Reply:
Wouldn't mp_poll use poll instead then?

Reply:
Sure but what would be the point? It's not like it would ever work. The main reason for it is for code used by linux/bsds where the BSD in question might not have ppoll. I think freebsd does, but not sure about the rest.